### PR TITLE
feat: prove Bitvec AddSub 1560

### PR DESCRIPTION
**Pull request by eurquhart1**
This fixes one more AliveStatement.

### CODE REVIEW

**Comment by eurquhart1** [comment]
Should these be moved to forLean.lean?

File: SSA/Projects/InstCombine/AliveStatements.lean [hunk 1]
```diff
@@ -114,13 +114,24 @@ theorem bitvec_AddSub_1556 :
...
 
+theorem allOnes_xor_eq_not (x : BitVec w) : allOnes w ^^^ x = ~~~x := by
+  apply eq_of_getLsb_eq
+  simp
+
+theorem BitVec.xor_comm : ∀ (e e_1: BitVec w), e ^^^ e_1 = e_1 ^^^ e := by
+  intros a b
+  apply eq_of_getLsb_eq
+  intros i
+  simp [getLsb_xor, Bool.xor_comm]
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/AliveStatements.lean:125]:
```suggestion
theorem xor_comm (e e_1 : BitVec w): e ^^^ e_1 = e_1 ^^^ e := by
  ext; simp [Bool.xor_comm]
```

Comment (tobiasgrosser) [reply SSA/Projects/InstCombine/AliveStatements.lean:125]: The `ext` tactic automates the `getLsb` application.

File: SSA/Projects/InstCombine/AliveStatements.lean [hunk 2]
```diff
@@ -114,13 +114,24 @@ theorem bitvec_AddSub_1556 :
...
 theorem bitvec_AddSub_1560 :
     ∀ (e : LLVM.IntW w), LLVM.sub (LLVM.const? (Int.negSucc 0)) e ⊑ LLVM.xor e (LLVM.const? (Int.negSucc 0)) := by
   simp_alive_undef
   simp_alive_ops
   simp_alive_case_bash
-  try alive_auto
-  all_goals sorry
+  simp
+  intro a
+  rw [negOne_eq_allOnes, allOnes_sub_eq_not, ← allOnes_xor_eq_not, BitVec.xor_comm]
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/AliveStatements.lean:134]: You probably want to add sth like to ForLean.lean:

```lean4
theorem allOnes_sub_eq_xor (x : BitVec w): (allOnes w) - x = x ^^^ (allOnes w) := by
  simp [negOne_eq_allOnes, allOnes_sub_eq_not, not_def, BitVec.xor_comm]
```

Our CI will break if you change AliveStatements.lean directly. For now maybe just add this definition to ForLean. I can then do an iteration to make sure it automatically applies to AliveStatements.lean.

**Comment by tobiasgrosser** [comment]
I cleaned this up for now. Let me close this.

**Pull request merged**